Formula/aws-sdk-cpp.rb

```diff
@@ -0,0 +1,90 @@
+require "set"
+
+class AwsSdkCpp < Formula
+  desc "AWS SDK for C++"
+  homepage "https://github.com/aws/aws-sdk-cpp"
+  url "https://github.com/aws/aws-sdk-cpp/archive/0.10.1.tar.gz"
+  sha256 "e571382dbccca0c37f38d097a12bf54874190a312e55ce22534cc6b1ddb8ffaf"
+  head "https://github.com/aws/aws-sdk-cpp.git"
+
+  WITHOUT_OPTIONS = {
+    "compute" => ["ec2", "elasticbeanstalk", "lambda"],
+    "storage" => ["s3", "cloudfront", "elasticfilesystem", "glacier", "importexport", "storagegateway"],
+    "database" => ["rds", "dynamodb", "elasticache", "redshift", "dms", "sdb"],
+    "networking" => ["directconnect", "route53", "route53domains"],
+    "devtools" => ["codecommit", "codedeploy", "codepipeline"],
+    "management" => ["cloudformation", "cloudtrail", "config", "opsworks", "ecr", "ecs", "autoscaling", "support", "logs", "monitoring", "ssm"],
+    "security" => ["identity-management", "access-management", "inspector", "waf", "iam", "cloudhsm", "acm", "ds", "kms", "sts"],
+    "analytics" => ["datapipeline", "kinesis", "firehose", "marketplacecommerceanalytics"],
+    "mobile" => ["cognito-identity", "cognito-sync", "devicefarm", "mobileanalytics", "sns"],
+    "appservices" => ["apigateway", "cloudsearch", "cloudsearchdomain", "elastictranscoder", "sqs", "swf", "email"],
+    "other" => ["elasticloadbalancing", "elasticmapreduce", "es", "events", "machinelearning", "queues", "transfer", "iot", "gamelift", "workspaces", "meteringmarketplace"],
+  }.freeze
+
+  WITHOUT_OPTIONS.each { |category, clients| option "without-#{category}", "Exclude clients: #{clients.join(", ")}" }
```

Comment:
@mikemcquaid I think you're right that many people will be fine with just the bottle. But I don't understand what the harm is in providing the options for people who'd like to do from-source builds. If you want to build static libs, you probably also only need a few of the clients. The difference between a 30 min build and a 3 min build is huge.

What is the big disadvantage of having these options here for those who need them? If some urgent need comes up to remove the options in the future, they could be removed and it wouldn't break anyone's setup, except for making their build take (potentially) 10x as long when it goes from just a few of the clients to many of the clients.